proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,1047 @@
+
+### Equivalent API for Web
+
+In parallel, a user-space layer over React DOM would ensure equivalent APIs between native and web, by converging them both towards a desired the "end state" where the differences between React Native and React DOM are erased while landing on a "modern" flavor of React DOM.
+
+The benefit of initially doing this in user-space is that it reduces the burden of complex migratations for React Native and React DOM (e.g., the changes to React DOM event types, the `style` prop, and explicit imports of component types), while allowing us to validate and iterate on the proposed end-state APIs with interested parties. This is currently blocked on OSS-ing of stylex.
+
+## How we teach this
+
+Teaching these APIs is simplified by developer familiarity with existing DOM / React DOM APIs.
+
+## Drawbacks
+
+Many of these features can be shimmed in user-space, at the cost of runtime overhead for native or web platforms. The runtime overhead for web is untenable. The runtime overhead of shifting shims into the React Native JS has yet to be established.
+
+This adds redundancy to the React Native API, as the proposal is to make additive changes and avoid requiring removal of existing APIs in the short term.
+
+We have to be very intentional about how much compatibility with open web standards we can promise. Some cross-platform limitations are rooted in significant differences in the host platforms APIs (e.g., accessibility) and the lack of 99% compat may frustrate users.
+
+Aiming for a degree of React DOM compatibility shifts the burden of missing features onto React Native. Developers may not be used to an API that is complete on web but partially implemented on native platforms.
+
+-----
+
+# Environment APIs
+
+## `window`
+
+### Events
+
+* [ ] Receives all W3C events dispatched to components (capture and bubble phase). Resolves proposal [#249](https://github.com/react-native-community/discussions-and-proposals/issues/249).
+* [ ] Dispatches [`resize`](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event).
+
+### Properties
+
+* [ ] [`devicePixelRatio`](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio)
+* [ ] [`navigator`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator)
+  * [ ] [`clipboard`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard)
+  * [ ] [`languages`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages)
+  * [ ] [`permissions`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/permissions)
+  * [ ] [`vibrate()`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/vibrate)
+* [`performance`](https://developer.mozilla.org/en-US/docs/Web/API/performance_property)
```

Comment:
Would be great if these were installed directly on the `window` object from native so we can reduce the amount of runtime JS required to create the expected runtime. Tons of shims results in more complicated bundling and larger bundles to debug.

Reply:
Isn't that what's proposed here?

"Tons of shims"... what shims are you referencing?